### DISCUSSION TITLE
Question about day of year climatology statistics

### DISCUSSION THREAD

**Discussion by huaracheguarache** [cf-convention/discuss#302] [Q&A about using CF] [answered]
### Question

I want to store climatological variables in a netCDF file that are calculated on a day-of-year basis for two different reference periods (1981-2010 and 1991-2020). All of the years in the data have been converted to leap years, and February 29th values have been interpolated where these are missing. I then take the subset of the data that is within the given reference period, and I group it based on the day of year and calculate the climatological variables. One example of such a variable is the median. The result is 366 values for each day of year of the given reference period.

I've been trying to look at chapter [7.4. Climatological Statistics](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#climatological-statistics) in the CF conventions to see whether it's possible to use the cell_methods and climatology_bounds defined there to store this data, but I'm not sure my data fits into what's specified there.

From what I understand, the standard only permits the following cell_methods:

* **time: method1** within years **time: method2** over years
* **time: method1** within days **time: method2** over days
* **time: method1** within days **time: method2** over days **time: method3** over years

Do any of these methods cover my use case? I don't think any of these cover things that are calculated on a day-of-year basis?

cc @TomLav

**Comment by davidhassell** [comment, marked as answer]
Thanks, @huaracheguarache, much appreciated! Here is the promised response: 

What is the characteristic of the daily values? If the daily data were means, then I think what you need would be 
`time: mean within years time: median over years`. The daily (i.e. "within years") aspect is encoded in the time coordinates and, crucially, their bounds:
```
data:  // time coordinates translated to date/time format
  time = "1981-01-01 12:00",
         "1981-01-02 12:00",
         ...
         "1981-12-31 12:00" ;
  climatology_bounds = "1981-01-01", "2010-01-02",
                       "1981-01-02", "2010-01-03",
                       ...
                       "1981-12-31", "2011-01-01" ;
```
Each pair of coordinate bounds spans the lower bound of the day in the first year to the upper bound of the same day in the last year. There is no flexibility on this. The time of day of the lower and upper bounds doesn't have to be midnight - they're whatever is defined in the original daily data.

The corresponding coordinate value is (anywhere) inside the day from the first year. It could also have been inside the same day from any of the years, e.g. any of the following `"1981-01-01 12:00"`, `"1982-01-01 12:00"`, ..., `"2010-01-01 12:00"` would be an acceptable coordinate value for the 1st January climatology.

I hope that helps,
David

**Reply by huaracheguarache** [reply]
Thanks for getting back to me! I'll say that they're daily means (even though that's not necessarily true, but it's not that important), which means that `time: mean within years time: median over years` would be correct. 

However, your answer raises a few questions. How can one differentiate between the cell method `time: median over years` meaning climatology on a day-of-year basis as opposed to a date basis? The two would produce different results, especially given a conversion to an all_leap calendar for all years and interpolation of February 29th values where these are missing.

Also, if I want to include two reference periods in the same file, would the way to proceed be to simply stack them on top of each other? By stacking I mean this:

```
data:  // time coordinates translated to date/time format
  time = "1981-01-01 12:00",
         ...
         "1981-12-31 12:00" ,
         "1991-01-01 12:00" ,
         ...
         "1991-12-31 12:00" ;
  climatology_bounds = "1981-01-01", "2010-01-02",
                       "1981-01-02", "2010-01-03",
                       ...
                       "1981-12-31", "2011-01-01" ,
                       "1991-01-01", "2020-01-02" ,
                       ...
                       "1991-12-31", "2021-01-01" ;
```

**Reply by huaracheguarache** [reply]
@davidhassell Bumping this.

**Comment by JonathanGregory** [comment]
Dear Michael @huaracheguarache 

Yes, if you have more than one climatology you can simply concatenate their climatological time coordinates, as you have suggested. It doesn't matter that they overlap. CF permits cells to overlap in general in any kind of coordinate.

The `within years` cell method refers to day of month within year. That's what you want, isn't it? I don't think we have considered using the day number 1,2,...,366 within the year instead. If there's a use case for it, I expect it could be done.

Best wishes

Jonathan

**Comment by huaracheguarache** [comment]
> Yes, if you have more than one climatology you can simply concatenate their climatological time coordinates, as you have suggested. It doesn't matter that they overlap. CF permits cells to overlap in general in any kind of coordinate.

Ok, great!

> The `within years` cell method refers to day of month within year. That's what you want, isn't it? I don't think we have considered using the day number 1,2,...,366 within the year instead. If there's a use case for it, I expect it could be done.

Actually, our use case is per day of year, not day of month. It's due to the way we have decided to deal with leap years in our data by converting all years to leap years and interpolating between February 28th and March 1st to get the value of February 29th for the non-leap years. After that we calculate the climatology on a day of year basis to produce a plot of sea ice data: https://ice.metsis-api.met.no/daily

**Comment by JonathanGregory** [comment]
Thanks for the explanation, @huaracheguarache. Yes, I understand that you interpolate. Having done that, you express the day within the year as DDMMM e.g. 28 Feb. That is what I mean by "day of month". Sorry not to be clear. It seems that the existing convention is suitable for your need, isn't it. Jonathan

**Reply by huaracheguarache** [reply]
No worries, I got things a bit mixed up myself. Yes, the convention would cover our needs. Thanks for your help!
